src/main/java/de/tum/in/www1/artemis/service/ExamService.java

```diff
@@ -115,4 +140,161 @@ public void delete(Long examId) {
+        for (var studentDto : studentDtos) {
+            var registrationNumber = studentDto.getRegistrationNumber();
+            try {
+                // 1) we use the registration number and try to find the student in the Artemis user database
+                Optional<User> optionalStudent = userService.findUserWithGroupsAndAuthoritiesByRegistrationNumber(registrationNumber);
+                if (optionalStudent.isPresent()) {
+                    var student = optionalStudent.get();
+                    // we only need to add the student to the course group, if the student is not yet part of it, otherwise the student cannot access the exam (within the course)
+                    if (!student.getGroups().contains(course.getStudentGroupName())) {
+                        userService.addUserToGroup(student, course.getStudentGroupName());
+                    }
+                    exam.addUser(student);
+                    continue;
+                }
+                // 2) if we cannot find the student, we use the registration number and try to find the student in the (TUM) LDAP, create it in the Artemis DB and in a potential
+                // external user management system
+                optionalStudent = userService.createUserFromLdap(registrationNumber);
+                if (optionalStudent.isPresent()) {
+                    var student = optionalStudent.get();
+                    // the newly created student needs to get the rights to access the course, otherwise the student cannot access the exam (within the course)
+                    userService.addUserToGroup(student, course.getStudentGroupName());
+                    exam.addUser(student);
+                    continue;
+                }
+                // 3) if we cannot find the user in the (TUM) LDAP, we report this to the client
+                log.warn("User with registration number " + registrationNumber + " not found in Artemis user database and not found in (TUM) LDAP");
+            }
+            catch (Exception ex) {
+                log.warn("Error while processing user with registration number " + registrationNumber + ": " + ex.getMessage(), ex);
+            }
+
+            notFoundStudentsDtos.add(studentDto);
+        }
+        examRepository.save(exam);
+        return notFoundStudentsDtos;
+    }
+
+    private List<Integer> assembleIndicesListWithRandomSelection(List<Integer> mandatoryIndices, List<Integer> optionalIndices, Long numberOfOptionalExercises) {
+        // Add all mandatory indices
+        List<Integer> indices = new ArrayList<>(mandatoryIndices);
+
+        // Add as many optional indices as numberOfOptionalExercises
+        if (numberOfOptionalExercises > 0) {
+            Collections.shuffle(optionalIndices);
+            indices = Stream.concat(indices.stream(), optionalIndices.stream().limit(numberOfOptionalExercises)).collect(Collectors.toList());
+        }
+
+        // Sort the indices
```

Comment:
Please clarify why you sort the indices.

Reply:
https://github.com/ls1intum/Artemis/pull/1678/commits/92df128e2ff65a9b2309ae9d663ddd9a8be2b791